### PR TITLE
Feature/scope change double column form

### PR DESCRIPTION
**Pull request by Gustav-Eikaas**
# Description

Please include a summary of the change and which issue is fixed.
List any external dependencies that are required for this change.

Completes:
[AB#200548](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/200548)
[AB#202499](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/202499)
[AB#202476](https://dev.azure.com/Equinor/fa63ceea-8883-41e2-8823-a3b54e4be178/_workitems/edit/202476)

## Checklist:

-   [x] I have performed a self-review of my own code.
-   [x] I have linked my DevOps task using the AB# tag.
-   [ ] My code is easy to read, and comments are added where needed.
-   [ ] My code is covered by tests.

### CODE REVIEW

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeBaseForm.tsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    });
+
+    return (
+        <BaseFormContainer>
+            <FormTextField
+                label="Title"
+                initialValue={state.title}
+                required
+                placeholder="Please add a title for the request"
+                onChange={(e) => handleInput('title', e)}
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeBaseForm.tsx:36]: change

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeBaseForm.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+                    onChange={(e) => handleInput('guesstimateDescription', e)}
+                    initialValue={state.guesstimateDescription}
+                />
+            </Guesstimate>
+        </BaseFormContainer>
+    );
+};
+
+interface FormTextFieldProps {
+    initialValue?: string;
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeBaseForm.tsx:113]: extract

File: src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestEditForm.tsx [hunk 3]
```diff
@@ -226,16 +205,18 @@ async function unpackRelatedObjects(
         options?: FetchQueryOptions<T, unknown, T, string[]> | undefined
     ) => Promise<T>
 ) {
+    function updateReferences(x: TypedSelectOption) {
+        const index = references.findIndex(({ value }) => value === x.value);
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/ScopeChangeRequestEditForm.tsx:209]: extract

File: src/apps/ScopeChangeRequest/Components/Form/useScopeChangeFormState.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            delete formData.current[key];
+        } else {
+            formData.current[key] = value as any;
+        }
+
+        const formState = checkFormState(formData.current);
+        if (isValid !== formState) {
+            setIsValid(formState);
+        }
+        setIsValid(checkFormState(formData.current));
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/useScopeChangeFormState.ts:44]: what

File: src/apps/ScopeChangeRequest/Components/Form/useScopeChangeFormState.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+                case !request.phase || request.phase.length <= 0:
+                    return false;
+
+                case !request?.changeCategory?.id || request?.changeCategory?.id.length <= 0:
+                    return false;
+
+                case !request.description || request.description.length <= 0:
+                    return false;
+
```
Comment (Gustav-Eikaas) [inline src/apps/ScopeChangeRequest/Components/Form/useScopeChangeFormState.ts:73]: cleanup

**Review by jaysencpp — APPROVED** [review]

**Pull request merged**